gen/cheader.tmpl

```diff
@@ -0,0 +1,234 @@
+{{- MCommentN .Copyright 0}}
+
+#ifndef {{.Name | ConstantCase}}_H_
+#define {{.Name | ConstantCase}}_H_
+
+#if defined(WGPU_SHARED_LIBRARY)
+#    if defined(_WIN32)
+#        if defined(WGPU_IMPLEMENTATION)
+#            define WGPU_EXPORT __declspec(dllexport)
+#        else
+#            define WGPU_EXPORT __declspec(dllimport)
+#        endif
+#    else  // defined(_WIN32)
+#        if defined(WGPU_IMPLEMENTATION)
+#            define WGPU_EXPORT __attribute__((visibility("default")))
+#        else
+#            define WGPU_EXPORT
+#        endif
+#    endif  // defined(_WIN32)
+#else       // defined(WGPU_SHARED_LIBRARY)
+#    define WGPU_EXPORT
+#endif  // defined(WGPU_SHARED_LIBRARY)
+
+#if !defined(WGPU_OBJECT_ATTRIBUTE)
+#define WGPU_OBJECT_ATTRIBUTE
+#endif
+#if !defined(WGPU_ENUM_ATTRIBUTE)
+#define WGPU_ENUM_ATTRIBUTE
+#endif
+#if !defined(WGPU_STRUCTURE_ATTRIBUTE)
+#define WGPU_STRUCTURE_ATTRIBUTE
+#endif
+#if !defined(WGPU_FUNCTION_ATTRIBUTE)
+#define WGPU_FUNCTION_ATTRIBUTE
+#endif
+#if !defined(WGPU_NULLABLE)
+#define WGPU_NULLABLE
+#endif{{"\n" -}}
+
+{{if eq .Name "webgpu"}}
+#include <stdint.h>
+#include <stddef.h>{{"\n" -}}
```

Comment:
I went through a pass of all `{{"\n" -}}` and removed all the ones that were unnecessary or replaceable.
Some are still there but for a reason, `{{"\n" -}}` were there to avoid unneeded newlines emitted from template. For example:

<table>
<tr>
<td>  </td>
<td> Template </td>
<td> Generates </td>
</tr>
<tr>
<td> 1. </td>
<td>

```go
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif{{"\n" -}}

{{if eq .Name "webgpu"}}
#include <stdint.h>
#include <stddef.h>
{{else}}
#include "webgpu.h"
{{end}}
```

</td>
<td>

```c
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif

#include <stdint.h>
#include <stddef.h>
```

</td>
</tr>

<tr>
<td> 2. </td>
<td>

```go
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif

{{if eq .Name "webgpu"}}
#include <stdint.h>
#include <stddef.h>
{{else}}
#include "webgpu.h"
{{end}}
```

</td>
<td>

```c
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif


#include <stdint.h>
#include <stddef.h>
```

</td>
</tr>

<tr>
<td> 3. </td>
<td>

```go
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif
{{if eq .Name "webgpu"}}
#include <stdint.h>
#include <stddef.h>
{{else}}
#include "webgpu.h"
{{end}}
```

</td>
<td>

```c
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif

#include <stdint.h>
#include <stddef.h>
```

</td>
</tr>

<tr>
<td> 4. </td>
<td>

```go
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif

{{- if eq .Name "webgpu"}}
#include <stdint.h>
#include <stddef.h>
{{else}}
#include "webgpu.h"
{{end}}
```

</td>
<td>

```c
#if !defined(WGPU_NULLABLE)
#define WGPU_NULLABLE
#endif
#include <stdint.h>
#include <stddef.h>
```

</td>
</tr>
</table>

Here only `1.` & `3.`  are producing desired the result, and `1.` allows us to have a newline in template, matching the generated result.